### PR TITLE
[Kubernetes] Update Kubernetes and KSM version for 1.28 compatibility

### PR DESCRIPTION
**Pull request by constanca-m**
## What

Supported Kubernetes versions are the ones being maintained [currently](https://endoflife.date/kubernetes): 1.26-1.28

For the `state_*` metricsets: supported versions are [2.8.2, 2.9.2 and 2.10](https://github.com/kubernetes/kube-state-metrics#compatibility-matrix). Steps:
1. Delete `_meta/test/ksm.v2.7.0` and add file for new supported version 2.10.0.
2. Update `docs.plain` to include this latest version.

For each `state_*` metricset:
1. Delete the old expected file.
2. Update the `files` variable under the test file.
3. Run `go test -data` to update the expected files.

For controller manager, proxy and scheduler:
1. Remove files for 1.25 version under `test`.
2. Add new file for version 1.28 and update `testdata/docs.plain` document with same content.

Additionally:
1. Update K8s version in `Jenkisfile`
2. Update K8s versions in `deploy/kubernetes/Jenkinsfile.yml`
3. Update docs: update `metricbeat/module/kubernetes/docs.asciidoc` with the new compatible versions and run `mage build` to update the `metricbeatbeat/docs/modules/kubernetes.asciidoc`

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

1. Clone this branch.
2. Run the tests for each metricset.

## Related issues

Relates to https://github.com/elastic/observability-dev/issues/2802.

### CODE REVIEW

File: metricbeat/module/kubernetes/_meta/test/docs/01_playground/prometheus.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
```
Comment (gizas) [inline metricbeat/module/kubernetes/_meta/test/docs/01_playground/prometheus.yaml:1]: Was this added by mistake?

Comment (constanca-m) [reply metricbeat/module/kubernetes/_meta/test/docs/01_playground/prometheus.yaml]: Yes! I removed it now

**Comment by gizas** [comment]
According to [this](https://github.com/elastic/beats/blob/main/docs/devguide/testing.asciidoc#generate-sample-events) can we run `go test --tags integration,kubernetes --data` and recreate all subfolder?

**Review by gizas — APPROVED** [review]

**Comment by constanca-m** [comment]
> According to [this](https://github.com/elastic/beats/blob/main/docs/devguide/testing.asciidoc#generate-sample-events) can we run go test --tags integration,kubernetes --data and recreate all subfolder?

I ran the tests for each separately, while I was updating the files

**Pull request merged**